src/modify.js

```diff
@@ -80,10 +80,22 @@ function rewriteAllFields(schema, configCallback, context) {
 
 export function modify(originalSchema, config) {
   const schema = JSON.parse(JSON.stringify(originalSchema));
+  let warn = []; // To be implemented in next PRs.
 
+  // All these functions mutate "schema",
+  // that's why we create a copy above
   rewriteFields(schema, config.fields);
 
   rewriteAllFields(schema, config.allFields);
 
-  return schema;
+  if (!config.muteWarningTip) {
+    console.warn(
+      'json-schema-form modify(): Make sure you log the returned `warn` as they highlight possible bugs in your modifications. To mute this log, pass `muteWarningTip: true` to the config.'
+    );
+  }
+
+  return {
+    schema,
+    warn: warn.length > 0 ? warn : undefined,
```

Comment:
Wouldn't `warnings` be a more suitable name? warn it's a verb and therefore a function, so it sounds a bit strange to me.

Nitpick: Why not simply return `warn` instead of running a ternary? Or alternatively, if you prefer returning a null value (instead of an empty array) we could initialize `warns` as `null` and set its type later in the code.

Reply:
Agree with the above, I think returning `warnings` at all times – empty if no warnings – would be expected.

I would still go with what I suggested in [this comment](https://github.com/remoteoss/json-schema-form/pull/80#discussion_r1687911026) though. It's non-blocking here since we can add the internal logging later, but it should affect how we name `muteWarningTip` and the existing default warning. If we go with internal logging, I would rename the property to `muteWarnings` or `muteLogging` since it's more generic, and remove the default log.

Reply:
To clarify on why keeping both internal logging and `warnings` array return:

- when using `modify` it's useful to get the warnings right away, no need for devs to do any additional config
- `warnings` return is useful to display these warnings in UI, e.g. json-schema-form own docs

Reply:
I thought about it but then as in the other PR https://github.com/remoteoss/json-schema-form/pull/80#discussion_r1684195117 was `warn` I let it be. I've renamed now to `warnings`.

But starting as `null` and then change it to array does not work well with mutations. Check this PR and run the tests, you'll see one failing.

Reassigining `warn` (now `warning`) from `null` to `[]` loses the array reference in the original value. and that's why it must be started as array. The alternative is to not use mutation. Instead return the warnings from each one of the mutating functions, like this:

```js
function modify(schema, config) {
        const splitResult = splitFields(schema, configOrder)
        const orderResult = reorderFields(schema, configOrder)
        // ... the same for all the other field mutations, create, remove, etc...

        const warns = [...splitResult.warn, ...orderResult.warn]
        
        return { 
          warn: warns.length > 0 ? warns : null
        }
}
```

After writing this I think I prefer the composable approach instead of mutation 😅

Reply:
Give me 2min, I'll show you the example in action right in this MR with the `rewriteFields` missing warning.

Reply:
Why do we want to return `warnings` as null though? Why not always return it and if no warnings it'll be an empty array ?

Reply:
Ups, took me longer, as I had to fight recursive warnings. Has tests too. Check it out! https://github.com/remoteoss/json-schema-form/pull/83/commits/42b31ca90ede3c4686f3d1effc45dcc0e234f244

Reply:
@danielcardoso5 so the devs can do `if (!warnings)` without nasty surprises as an empty array being considered truthy 🫠

Reply:
> nasty surprises as an empty array being considered truthy

Why would this be bad? The name suggests an array.

@sandrina-p could you address [these comments above](https://github.com/remoteoss/json-schema-form/pull/83#discussion_r1689581954)? Especially the naming `muteWarningTip` since it affects a prop name that might change.

Reply:
> Why would this be bad? The name suggests an array.

Because based on experience, devs forget about this simple thing. But no strong opinion, so yeah, empty array it is. changed.

Reply:
I see, although I think here it's quite safe. I expect `warnings` to only be used to either log existing warning messages (`warnings.forEach`) – if we don't add internal logging – or to display the warnings in UI ( `warnings.map`), empty array would be okay in both occasions IMO.

I think we just need an updated PR description and we're good to go! 🚀 I'll approve

Reply:
Yes, it's minimal. PR description updated :D